frontend/src/components/DirectoryInfoModal.tsx

```diff
@@ -0,0 +1,137 @@
+/* eslint-disable @typescript-eslint/no-misused-promises */
+import { zodResolver } from "@hookform/resolvers/zod";
+import React from "react";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+import "./DirectoryInfoModal.css";
+
+const directoryInfoSchema = z.object({
+  degree: z.string().min(3, "Degree must be at least 3 characters"),
+  institution: z.string().min(3, "Institution must be at least 3 characters"),
+  clinic: z.string().optional(),
+  website: z.string().url("Invalid website URL").optional(),
+  country: z.string().nonempty("Country is required"),
+  addressLine: z.string().optional(),
+  apartment: z.string().optional(),
+  city: z.string().optional(),
+  state: z.string().optional(),
+  postcode: z.string().min(5, "Postcode must be at least 5 characters"),
+});
+
+type DirectoryInfoFormData = z.infer<typeof directoryInfoSchema>;
+
+interface DirectoryInfoModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  onSubmit: (data: DirectoryInfoFormData) => void;
+}
+
+const DirectoryInfoModal: React.FC<DirectoryInfoModalProps> = ({ isOpen, onClose, onSubmit }) => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm<DirectoryInfoFormData>({
+    resolver: zodResolver(directoryInfoSchema),
+  });
+
+  if (!isOpen) return null;
+
+  const submitForm = (data: DirectoryInfoFormData) => {
+    onSubmit(data);
+    reset();
+    onClose();
+  };
+
+  return (
+    <div className="modal-overlay">
+      <div className="modal-container">
+        <button className="close-button" onClick={onClose}>
+          &times;
+        </button>
+        <h2>Edit Directory Info</h2>
```

Comment:
```suggestion
        <h2>Edit Personal Info</h2>
```

lets match the design text